packages/tkg-clusterclass-aws/bundle/config/upstream/overlay-pod-security-admission.yaml

```
@@ -64,71 +64,104 @@ spec:
-              kind: AdmissionConfiguration
-              plugins:
-              - name: PodSecurity
-                configuration:
-                  apiVersion: pod-security.admission.config.k8s.io/v1beta1
-                  kind: PodSecurityConfiguration
-                  defaults:
-                    enforce: "{{ if .podSecurityStandard.enforce -}}
-                        {{ .podSecurityStandard.enforce }}
-                      {{- end }}"
-                    enforce-version: "{{ .podSecurityStandard.enforceVersion -}}"
-                    audit: "{{ if .podSecurityStandard.audit -}}
-                        {{ .podSecurityStandard.audit }}
-                      {{- else -}}
-                        {{ $defaultWarnAudit }}
-                      {{- end }}"
-                    audit-version: "{{ .podSecurityStandard.auditVersion -}}"
-                    warn: "{{ if .podSecurityStandard.warn -}}
-                        {{ .podSecurityStandard.warn }}
-                      {{- else -}}
-                        {{ $defaultWarnAudit }}
-                      {{- end }}"
-                    warn-version: "{{ .podSecurityStandard.warnVersion -}}"
-                  exemptions:
-                    usernames: []
-                    runtimeClasses: []
-                    namespaces: [{{ $namespace_exemptions }}]
-            path: /etc/kubernetes/kube-apiserver-admission-pss.yaml
-    enabledIf: '{{ and (not .podSecurityStandard.deactivated) (semverCompare ">= v1.24" .builtin.controlPlane.version) }}'
+  - name: apiserverEventRateLimitConf
+    required: false
+    schema:
+      openAPIV3Schema:
+        type: string
+        default: ""
+  #@overlay/match missing_ok=True
+  patches:
+    #@overlay/remove
+    #@overlay/match by=overlay.map_key("name"),missing_ok=True
+    - name: admissionConfiguration
+    #@overlay/append
+    - name: admissionConfiguration
+      description: "Adds an admission configuration for kube-apiserver."
+      definitions:
+        - selector:
+            apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+            kind: KubeadmControlPlaneTemplate
+            matchResources:
+              controlPlane: true
+          jsonPatches:
+            - op: add
+              path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraArgs/admission-control-config-file"
+              valueFrom:
+                template: |
+                  {{ $enableAdmissionPlugin := false }}
+                  {{- range $key, $val := .apiServerExtraArgs }}
+                  {{- if eq $key "enable-admission-plugin" }}
+                  {{- $enableAdmissionPlugin = true }}
+                  {{- end }}
+                  {{- if not $enableAdmissionPlugin }}
+                  enable-admission-plugins: "NodeRestriction,EventRateLimit"
+                  {{- end }}
+                  admission-control-config-file: "/etc/kubernetes/admission-control-config.yaml"
```

The args need to get separated, one OP per extraArg. Currently this would add a single extraArg called `admission-control-config-file` (see the last part of the path key) and add a multi-line string as value.

Regarding the `enable-admission-plugins`: I think this needs to get separated to an separate patch (element in `.spec.topology.patches`). I think you not always want to add the flag right? This could only be done by using the `enabled_if` in a patch.
```suggestion
            - op: add
              path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraArgs/admission-control-config-file"
              value: "/etc/kubernetes/admission-control-config.yaml"
```